sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -775,14 +791,25 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
         convert(And(GreaterThanOrEqual(child, Literal(sortedValues.head, dataType)),
           LessThanOrEqual(child, Literal(sortedValues.last, dataType))))
 
+      case Not(InSet(_, values)) if values.size > inSetThreshold =>
```

Comment:
can we move this to the beginning so that unsupported cases are grouped together?

Reply:
moved.